### ISSUE TITLE
iOS Crush

### ISSUE THREAD

**Issue by kyleap** [phonegap/phonegap-plugin-barcodescanner#220]
I used ionic freamwork , but the screen was freeze when i open the camara,
Android is work fine . hear is my code , any suggests ? thanks 

version:phonegap-plugin-barcodescanner 4.1.0

$ionicPlatform.ready(function() { 
　　$scope.scanBarcode = function() { 
　　　　$cordovaBarcodeScanner.scan().then(function(result) { 
　　　　　// result 
　　　　}, function(error) { $log.debug("An error happened -> " + error); }); 
　　}; 
});

**Comment by gaudecker** [comment]
We've had the same issue for almost a year now for our iOS app.

**Comment by gaudecker** [comment]
@kyleap I fixed this issue by forking the plugin and fixing the issue like this: silmu/phonegap-plugin-barcodescanner@7b8d05cc1a1d0b0a6b94593a0511c6089bd8b81b

Then replacing `phonegap-plugin-barcodescanner` under `cordovaPlugins` in package.json with the url to our fork:

```json
{
  "id": "phonegap-plugin-barcodescanner",
  "locator": "https://github.com/silmu/phonegap-plugin-barcodescanner.git"
},
```

**Comment by kyleap** [comment]
@gaudecker thank you for your help :)

**Comment by EddyVerbruggen** [comment]
A few of my plugins are also affected by this Ionic bug on iOS so I rather see that fixed than fixing all plugins, but I'm happy to see (and consider) these workarounds 👍

**Comment by nikhilkh** [comment]
@EddyVerbruggen I'm curious - what is the Ionic bug on iOS here? Is that logged already - Is scan being invoked twice for some reason?

**Comment by EddyVerbruggen** [comment]
@nikhilkh https://github.com/driftyco/ionic/issues/1022

**Comment by pke** [comment]
@EddyVerbruggen true, the source bug needs to be fixed. But this plugin can never be sure another framework (or manually) calls the scan function twice. If calling it twice crashes the app then the plugin should protect itself and drop out early of double scan calls.

**Comment by EddyVerbruggen** [comment]
Or your app should protect against it. Like I said: willing to consider a workaround (PR).

**Comment by pke** [comment]
@gaudecker mentioned 2 workarounds already.
I stand by that API should protect itself against (accidental) wrong calls.

**Comment by EddyVerbruggen** [comment]
@pke you're welcome to submit a PR

**Comment by EddyVerbruggen** [comment]
@daserge LGTM

**Comment by daserge** [comment]
Tested with ionic and ionic-material and the issue does not reproduce.

**Comment by daserge** [comment]
Fixed in #267
